### ISSUE TITLE
Improving the notifying process

### ISSUE THREAD

**Issue by justinpark** [toranb/ember-redux#37]
Hello @toranb 
Thank you for your solution.
Our team love your solution and try to apply into the internal solution.
I found one thing to fix to improve the observing performance.

I'm not a good writer. Please understand the lack of description.

The problem is the `connect` function notifies all mapped properties even if only one of them has been updated.

For example, I have one container component that watches the router state and a collection record.
```javascript
function select(store) {
  const {
    collection,
    router,
  } = this.store;
  return {
    collection,
    router,
  }
}
```
And then the component observes the router state and updates the collection set once the collection filter parameter is updated.
```
...
onRouteChange: Ember.observer('router', function() {
   const redux = this.get('redux');
   redux.dispatch({type: 'UPDATE_COLLECTION', payload: {filter: router.params}});
})
```
The router observer has been triggered after the collection state is updated because the `handleChange` compares the all states mapped and triggers the `notifyPropertyChange` all the states keys. This makes the infinite loop and breaks the system.
```javascript
                var props = mapState(redux.getState());
...
                if (!shallowEqual(componentState, reduxState)) {
                    //the following process notifies all props
                    this.updateProps(props);
                }
```

# Suggestion
We solved the problem by notifying the only updated state keys.
On `connect` side, we would change the notifying process only updated values.
```javascript
            handleChange() {
                var redux = this.get('redux');
                var props = mapState(redux.getState());
                var componentState = this.getComponentState(props);
                var reduxState = finalMapStateToComputed(redux.getState());
                props.forEach((name) => {
                  if (!shallowEqual(componentState[name], reduxState[name])) {
                    this.notifyPropertyChange(name);
                  }
                });
            },
```

How do you think of my idea?
If the suggestion works, please patch the change on the next version.

Thanks.

Regards,
@justinpark

**Comment by toranb** [comment]
Good news :) we already have a PR in flight to solve the problem you mention

https://github.com/toranb/ember-redux/pull/30

**Comment by toranb** [comment]
To push this ahead much quicker, I've opened a PR solely focused on the over notify problem you bring up here. 

https://github.com/toranb/ember-redux/pull/38

My hope is we can get a release out this weekend -sorry for the delay!

**Comment by toranb** [comment]
@justinpark I just published v1.8.0 w/ this change. Please pull this down and confirm when you have time :)
